plugin/core/sessions.py

```diff
@@ -875,6 +876,17 @@ def on_pre_send_notification_async(self, notification: Notification) -> None:
         """
         pass
 
+    def on_server_response(self, method: str, result: Any) -> None:
```

Comment:
I don't have the mind for it right now but I would want to think about whether this is generic enough approach and how would it fit in the future if we'd potentially want to also hook into other things like:
- hooking server initiated notification/request
- hooking client initiated notification/request
- hooking client response to server initiated request.

Reply:
So do you mean to only have a single method for all different kinds, like `on_payload`? I think that would make it more complicated with more parameters like "direction", and plugins would then need to do everything what LSP already does in `Session.deduce_payload` themselves. And there are already a few methods in AbstractPlugin:

> * hooking client-initiated notification/request

This should already be possible with the `on_pre_send_*` methods.

I guess if we need more methods e.g. to modify incoming notifications before passing them on to LSP, we could add it on demand when we find a usecase for it.

Reply:
I'm not sure yet. I would have to think about it and maybe prototype a bit. Would not want to end up with inconsistent and confusing API if we need to add stuff later.

Since we have `on_pre_send_request_async` and `on_pre_send_notification_async` already, we should probably either be as consistent with those as possible or deprecate those and create something more generic.

Not sure yet.